src/services/completions.ts

```diff
@@ -2125,6 +2127,21 @@ export function getCompletionEntriesFromSymbols(
                 return false;
             }
 
+            if (parameterDeclaration) {
+                // Filter out parameters from their own initializers
+                // `function f(a = /* no 'a' here */) { }`
+                if (symbol.valueDeclaration === parameterDeclaration) {
+                    return false;
+                }
+                // Filter out parameters from other parameters' initializers
+                // `function f(a = /* no 'b' here */, b) { }`
+                const parameters = parameterDeclaration.parent.parameters;
+                const currentParamIdx = parameters.indexOf(parameterDeclaration);
+                if (parameters.slice(currentParamIdx).some((p) => symbol.valueDeclaration === p)) {
+                    return false;
+                }

```

Comment:
I wonder if you could just get await with using the node positions.

```suggestion
                const declarationPos = symbol.valueDeclaration.pos;
                if (parameterDeclaration.pos <= declarationPos && declarationPos < parameters.end) {
                    return false;
                }
```

Reply:
Seems to be good idea! but I still wonder why do we need `declarationPos < parameters.end` check?